controller/organization.go

```diff
@@ -0,0 +1,161 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-auth/account"
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/authorization/resource"
+	"github.com/fabric8-services/fabric8-auth/authorization/role"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/goadesign/goa"
+	"github.com/satori/go.uuid"
+	"strings"
+)
+
+const OrganizationOwnerRole = "owner"
+
+// OrganizationController implements the organization resource.
+type OrganizationController struct {
+	*goa.Controller
+	db           application.DB
+	TokenManager token.Manager
+}
+
+// NewOrganizationController creates an organization controller.
+func NewOrganizationController(service *goa.Service, db application.DB) *OrganizationController {
+	return &OrganizationController{Controller: service.NewController("OrganizationController"), db: db}
+}
+
+// Create runs the create action.
+func (c *OrganizationController) Create(ctx *app.CreateOrganizationContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+
+	if len(strings.TrimSpace(*ctx.Payload.Name)) == 0 {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest("Organization name cannot be empty"))
+	}
+
+	var organizationId uuid.UUID
+
+	err = application.Transactional(c.db, func(appl application.Application) error {
+
+		// Lookup the identity for the current user
+		userIdentity, err := appl.Identities().Load(ctx, *currentUser)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewInternalErrorFromString(ctx, "Error looking up current user"))
+		}
+
+		// Lookup the organization resource type
+		resourceType, err := appl.ResourceTypeRepository().Lookup(ctx, account.IdentityResourceTypeOrganization)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewInternalErrorFromString(ctx, "Error looking up resource type 'identity/organization'"))
```

Comment:
Let's not cast the error to an internal error because `Lookup` might return an `error` even if the record was not found in the database 
https://github.com/fabric8-services/fabric8-auth/pull/317/files#diff-76f758602c8a7d4581a1e3e77b1bda3fR217
in that case, the error should get reported as NotFound and not InternalError
So, I suggest we keep it as 
```
return jsonapi.JSONErrorResponse(ctx, err)
```

Sorry, didn't notice this earlier.

Reply:
Done.